js/foundation.offcanvas.js

```diff
@@ -118,10 +118,16 @@ class OffCanvas extends Plugin {
       }
     }
 
-    this.options.isRevealed = this.options.isRevealed || new RegExp(this.options.revealClass, 'g').test(this.$element[0].className);
+    // Get the revealOn option from the class.
+    var revealOnClass = this.$element[0].className.match(RegExpEscape(this.options.revealClass) + '([^\s])+', 'g');
```

Comment:
match() does not work this way – you must either use a hard coded pattern or a RegExp pattern if you need variables in it.
So it has to look like this
```javascript
var revealOnClass = this.$element[0].className.match(new RegExp(RegExpEscape(this.options.revealClass) + '([^\s])+', 'g'));
```

Reply:
Fixed